src/Knp/Rad/ResourceResolver/ParameterCaster/VariableCaster.php

```diff
@@ -0,0 +1,36 @@
+<?php
+
+namespace Knp\Rad\ResourceResolver\ParameterCaster;
+
+use Knp\Rad\ResourceResolver\ParameterCaster;
+use Symfony\Component\HttpFoundation\RequestStack;
+
+class VariableCaster implements ParameterCaster
+{
+    private $request;
+
+    public function __construct(RequestStack $requestStack)
+    {
+        $this->request = $requestStack->getCurrentRequest();
+    }
+
+    public function supports($string)
+    {
+        if (preg_match('/^\$[a-zA-Z_]+/', $string)) {
+            return true;
+        } else {
+            return false;
+        }
+    }
+
+    public function cast($string)
+    {
+        $routeParameters = $this->request->attributes->get('_route_params');
+
+        foreach ($routeParameters as $routeParameter => $value) {
+            if (substr($string, 1) === $routeParameter) {
+                return $value;
+            }
+        }
```

Comment:
And what is the behavior if there is no variable available ? For exemple I've got a `$test` but no `test` attribute in the request.